app/src/main/java/io/appium/settings/RecorderThread.java

```diff
@@ -0,0 +1,437 @@
+        encoderFormat.setInteger(MediaFormat.KEY_I_FRAME_INTERVAL,
+                RecorderConstant.AUDIO_CODEC_I_FRAME_INTERVAL_MS);
+        return encoderFormat;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private VirtualDisplay initVirtualDisplay(MediaProjection mediaProjection,
+                                               Surface surface, Handler handler,
+                                               int videoWidth, int videoHeight) {
+        return mediaProjection.createVirtualDisplay("Appium Screen Recorder",
+                videoWidth, videoHeight, DisplayMetrics.DENSITY_HIGH,
+                DisplayManager.VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR,
+                surface, displayCallback, handler);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private MediaCodec initAudioCodec(int sampleRate) throws IOException {
+        // Encoded video resolution matches virtual display. TODO set channelCount 2 try stereo quality
+        MediaFormat encoderFormat = MediaFormat.createAudioFormat(MediaFormat.MIMETYPE_AUDIO_AAC,
+                sampleRate, RecorderConstant.AUDIO_CODEC_CHANNEL_COUNT);
+        encoderFormat.setInteger(MediaFormat.KEY_BIT_RATE,
+                RecorderConstant.AUDIO_CODEC_DEFAULT_BITRATE);
+
+        MediaCodec audioEncoder = MediaCodec.createEncoderByType(MediaFormat.MIMETYPE_AUDIO_AAC);
+        audioEncoder.configure(encoderFormat, null, null,
+                MediaCodec.CONFIGURE_FLAG_ENCODE);
+        return audioEncoder;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.Q)
+    private AudioRecord initAudioRecord(MediaProjection mediaProjection, int sampleRate) {
+        int channelConfig = AudioFormat.CHANNEL_IN_MONO;
+        int minBufferSize = AudioRecord.getMinBufferSize(sampleRate, channelConfig,
+                AudioFormat.ENCODING_PCM_16BIT);
+
+        AudioFormat audioFormat = new AudioFormat.Builder()
+                .setEncoding(AudioFormat.ENCODING_PCM_16BIT)
+                .setSampleRate(sampleRate)
+                .setChannelMask(AudioFormat.CHANNEL_IN_MONO)
+                .build();
+
+        AudioRecord.Builder audioRecordBuilder = new AudioRecord.Builder();
+        AudioPlaybackCaptureConfiguration apcc =
+                new AudioPlaybackCaptureConfiguration.Builder(mediaProjection)
+                        .addMatchingUsage(AudioAttributes.USAGE_MEDIA)
+                        .build();
+        return audioRecordBuilder.setAudioFormat(audioFormat)
+                .setBufferSizeInBytes(4 * minBufferSize)
+                .setAudioPlaybackCaptureConfig(apcc)
+                .build();
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private Thread startAudioRecord(MediaCodec audioEncoder, final AudioRecord audioRecord) {
+        return new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Thread.currentThread().setPriority(Thread.MAX_PRIORITY);
```

Comment:
this should also probably be configurable. For some people it is more important to have the app under test performing well rather than the video recording

Reply:
done https://github.com/sirmordred/io.appium.settings/commit/d229c009f8bb8e9e00f4dbe70f99b890e3b1aaf7